### PR TITLE
Add logging interface

### PR DESCRIPTION
**Pull request by sadmansk**
**Goal:** Add a static logger that can be used from anywhere in the program.  
**Features:**
* Able to set custom destinations (different Qt Text Boxes)
* Current time displayed with the log message
* Different levels of log messaging, `ERROR` and `INFO` level for this iteration
* Different colors for each level of logging

End Result:
logging

### CODE REVIEW

File: code/gui/mainwindow.ui [hunk 1]
```diff
@@ -80,16 +80,27 @@
...
       </widget>
      </item>
      <item>
-      <widget class="QTextBrowser" name="textBrowser"/>
+      <widget class="QTextEdit" name="log_viewer">
+       <property name="toolTip">
+        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Log Console&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
+       </property>
+       <property name="html">
+        <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
```
Comment (aneedalie) [inline code/gui/mainwindow.ui]: Could we format this better? If this is an automatically generated file then nevermind.

File: code/main.cpp [hunk 2]
```diff
@@ -7,13 +7,12 @@ int main(int argc, char *argv[])
     QApplication app(argc, argv);
     MainWindow *w = new MainWindow();
 
-    Logger *logger = new Logger();
-    w->show();
+    Logger::SetStream(w->GetLogView());
+    Logger::Log("Logger initialized.");
+
+    Logger::Log("ERROR: No controllers were found!", Logger::ERROR);
```
Comment (aneedalie) [inline code/main.cpp]: Going to assume this is for demo/testing purposes and wouldn't normally be hardcoded in.

**Comment by aneedalie** [comment]
:+1: everything looks good :smiley: :gem: :100:

File: code/main.cpp (hunk 2 above)
Comment (sadmansk) [inline code/main.cpp]: Yeap, I will be taking it out, probably move both the statements into their corresponding initializers/ctors.

File: code/gui/mainwindow.ui (hunk 1 above)
Comment (sadmansk) [inline code/gui/mainwindow.ui]: It is automatically generated but I will be naming the objects appropriately as I go, and also remove the ones that we dont want. Currently we have the exact same GUI as the python base, but I'm against keeping all of those things.

**Comment by sadmansk** [comment]
@aneedalie thanks a lot tammy, merging.

**Pull request merged**